### PR TITLE
Dockerize app

### PR DESCRIPTION
**Pull request by timstallmann**
@allthesignals this PR accomplishes the following:

* Remove unused crontab and sinatra infrastructure from the repo
* Add a .gitignore file and remove some other unneeded repo files
* Finalizes a working docker infrastructure for the app, encompassing a ruby container for the app (which is spun up to run a single command, either to scrape or set-up/migrate the DB), and a PostGIS container for the DB. By default, DB storage is persistent only as long as the PostGIS container is running, but a volume mapping can be set up in the docker-compose file to enable persistent DB storage locally.
* Adds a makefile to create shorthands for basic commands (setting up a fresh DB, running the scraper, exporting data to GeoJSON)
* Adds a `uid` column to the `listings` table which stores either the Craigslist URL or the PadMapper listing ID.. this column can be used to query for unique listings, but it does not currently impose any uniqueness constraints on the DB itself
* Adds some more reporting
* Abstracts some of the Boston-specific constants into environment variables
* Adds a `MAX_RESULTS` environment variable so that when you're testing the scraper locally in development it will pull a small number of results and then exit gracefully

Some things which this PR does *not* do:
* Better geojson output -- the current version just dumps the entire listings table to geojson, which will quickly get unwieldy because of the lack of deduplication across multiple scraper runs. It would be nice to add some functionality to pull listings from a particular scraper run, or from a particular date
* Deduplication of any form -- I plan to experiment with the basic dedupe stuff we discussed a bit on a separate branch and see how it meets our needs in Durham, but still unsure about how it will work with real-world listings
* More clear deployment instructions

### CODE REVIEW

File: mux/craigslist.rb [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
     sleep(5)
-    geocluster_url = "http://boston.craigslist.org" + geocluster
+    geocluster_url = @@base_url + geocluster
     uri = URI(geocluster_url)
     res = Net::HTTP.get_response(uri)
     results = JSON.parse( assert_successful_response (res) ).first
 
     results.each do |r|
-      create_listing_from_result(r, survey) if !r.has_key?("GeoCluster")
+      create_listing_from_result(r, survey) unless r.has_key?('GeoCluster')
```
Comment (allthesignals) [inline mux/craigslist.rb:29]: Great

**Review by allthesignals — APPROVED** [review]
Thank you for adding documentation as well.

**Comment by allthesignals** [comment]
This is fantastic and looks great to me - thank you, @timstallmann. I will add a few issues so we can keep track of things you and I are noticing.

**Comment by allthesignals** [comment]
Hi @timstallmann, I just had a quick question about https://github.com/MAPC/rental-listing-aggregator/blob/master/docker-compose.yml#L24. 

I'd like to try this on Heroku, and I need the environment to point to a remote database. I don't see these variables referenced anywhere else in the source, so I wonder if this is Docker Compose-specific? I'm assuming that I can just add `POSTGRES_HOST`, for example - unless I need to wire up this new variable in a database_url somewhere. 

Thanks!

**Comment by timstallmann** [comment]
Yeah! The database settings for the ruby app are all in config/database.yml, so you would set the `POSTGRES_HOST` there and then just comment out the whole `db` service in the Docker compose file since there's no need to spin up a PostGIS db from docker if you're using a remote DB.

**Pull request merged**